### ISSUE TITLE
Can't update Spyder from with Spyder 6.0.1

### ISSUE THREAD

**Issue by joooeey** [spyder-ide/spyder#23157]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide (**So far, I have not tried uninstalling Spyder**)
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

The built-in update from Spyder 6.0.1 doesn't succeed.

### What steps reproduce the problem?

1. Open Spyder.
2. Wait for the dialog that says "Spyder 6.0.2 is available. Would you like to install it?"
3. Press "After closing" or "Yes". In the former case, you also have to close Spyder.

### What is the expected output? What do you see instead?

I expect an update to Spyder 6.0.2 but it fails. See the Traceback below.

### Paste Traceback/Error Below (if applicable)

```python-traceback

=========================================================
Updating Spyder
---------------

IMPORTANT: Do not close this window until it has finished
=========================================================

Waiting for Spyder to quit...
Waiting for Spyder to quit...
Spyder quit.
/tmp/spyder-lukas/updates/6.0.2 ~/Desktop/repos/rqa
Updating Spyder base environment...

CondaFileIOError: 'conda-base-linux-64.lock'. [Errno 2] No such file or directory: 'conda-base-linux-64.lock'

Updating Spyder runtime environment...

CondaFileIOError: 'conda-runtime-linux-64.lock'. [Errno 2] No such file or directory: 'conda-runtime-linux-64.lock'

Cleaning packages and temporary files...
There are no unused package(s) to remove.
There are no tempfile(s) to remove.
Press return to exit and launch Spyder...  

```

## Versions

* Spyder version: 6.0.2
* Python version: 3.12.7 64-bit
* Qt version: 5.15.8
* PyQt version: 5.15.9
* Operating System name/version: Linux 5.15.0-122-generic (x86_64) a.k.a. Linux Mint

### Dependencies

```
# Mandatory:
aiohttp >=3.9.3                  :  3.10.10 (OK)
asyncssh >=2.14.0,<3.0.0         :  2.17.0 (OK)
atomicwrites >=1.2.0             :  1.4.1 (OK)
chardet >=2.0.0                  :  5.2.0 (OK)
cloudpickle >=0.5.0              :  3.0.0 (OK)
cookiecutter >=1.6.0             :  2.6.0 (OK)
diff_match_patch >=20181111      :  20230430 (OK)
github >=2.3.0                   :  2.4.0 (OK)
importlib_metadata >=4.6.0       :  8.5.0 (OK)
intervaltree >=3.0.2             :  3.1.0 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1 :  8.28.0 (OK)
jedi >=0.17.2,<0.20.0            :  0.19.1 (OK)
jellyfish >=0.7                  :  1.1.0 (OK)
jsonschema >=3.2.0               :  4.23.0 (OK)
keyring >=17.0.0                 :  25.4.1 (OK)
nbconvert >=4.0                  :  7.16.4 (OK)
numpydoc >=0.6.0                 :  1.8.0 (OK)
parso >=0.7.0,<0.9.0             :  0.8.4 (OK)
pexpect >=4.4.0                  :  4.9.0 (OK)
pickleshare >=0.4                :  0.7.5 (OK)
psutil >=5.3                     :  6.0.0 (OK)
pygments >=2.0                   :  2.18.0 (OK)
pylint >=3.1,<4                  :  3.3.1 (OK)
pylint_venv >=3.0.2              :  3.0.3 (OK)
pyls_spyder >=0.4.0              :  0.4.0 (OK)
pylsp >=1.12.0,<1.13.0           :  1.12.0 (OK)
pylsp_black >=2.0.0,<3.0.0       :  2.0.0 (OK)
pyuca >=1.2                      :  1.2 (OK)
qdarkstyle >=3.2.0,<3.3.0        :  3.2.3 (OK)
qstylizer >=0.2.2                :  0.2.3 (OK)
qtawesome >=1.3.1,<1.4.0         :  1.3.1 (OK)
qtconsole >=5.6.0,<5.7.0         :  5.6.0 (OK)
qtpy >=2.4.0                     :  2.4.1 (OK)
rtree >=0.9.7                    :  1.3.0 (OK)
setuptools >=49.6.0              :  75.1.0 (OK)
sphinx >=0.6.6                   :  8.1.0 (OK)
spyder_kernels >=3.0.0,<3.1.0    :  3.0.0 (OK)
superqt >=0.6.2,<1.0.0           :  0.6.7 (OK)
textdistance >=4.2.0             :  4.6.2 (OK)
three_merge >=0.1.1              :  0.1.1 (OK)
watchdog >=0.10.3                :  5.0.3 (OK)
xdg >=0.26                       :  0.28 (OK)
yarl >=1.9.4                     :  1.14.0 (OK)
zmq >=24.0.0                     :  26.2.0 (OK)

# Optional:
cython >=0.21                    :  None (NOK)
matplotlib >=3.0.0               :  3.9.2 (OK)
numpy >=1.7                      :  2.1.2 (OK)
pandas >=1.1.1                   :  2.2.2 (OK)
scipy >=0.17.0                   :  1.14.1 (OK)
sympy >=0.7.3                    :  None (NOK)
```

**Comment by ccordoba12** [comment]
Hey @joooeey, thanks for reporting. Some comments about what you said:

> I expect an update to Spyder 6.0.2 but it fails. See the Traceback below.

How did you install Spyder? It's not clear if you used conda, pip or our standalone installer.

BTW, if you used the Spyder package that comes with your Linux distro, then the update process to go from 6.0.1 to 6.0.2 won't work. That was fixed in 6.0.2 by simply not offering the update in that case.

> CondaFileIOError: 'conda-base-linux-64.lock'. [Errno 2] No such file or directory: 'conda-base-linux-64.lock'

It seems Spyder was unable to download or write to disk the file necessary to perform the update. Please check that the directory where you installed Spyder has the right permissions.

> Versions
> Spyder version: 6.0.2

Your current version (the one attached automatically to this issue by Spyder) is 6.0.2. So, how did you perform the update at the end?

**Comment by joooeey** [comment]
> Your current version (the one attached automatically to this issue by Spyder) is 6.0.2. So, how did you perform the update at the end?

That was a typo. I copied the number by hand. I had version 6.0.2 .

> How did you install Spyder? It's not clear if you used conda, pip or our standalone installer.

It wasn't clear to me either. I assumed stand-alone but apparently I installed with `mamba install spyder` into a specific mamba environment. Consequently I managed to do the update with `mamba activate roads` and `mamba update spyder` just now.

I'm closing this issue now but clearer language in the nagging update dialog or in the traceback would be helpful.
